### PR TITLE
criocli: fix/diff-friendlier `zsh` completion generation.

### PR DESCRIPTION
**Pull request by klihub**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Commands with a single quote (`'`) in their usage string cause syntactically incorrect `zsh` completion generation. This patchset aims to fix that, while it also reformats the generated `zsh` completions for easier diffing with less contextual noise in the future.

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
Fix zsh completion generation.
```

### CODE REVIEW

**Comment by haircommander** [comment]
/ok-to-test
/approve

LGTM, thanks @klihub

**Comment by haircommander** [comment]
@cri-o/cri-o-maintainers PTAL

**Review by fidencio — COMMENTED** [review]
@klihub, just a small suggestion, which could be applied to pretty much all blocks changed.
Please, let me know what do you think.

File: completions/zsh/_crio [hunk 1]
```diff
@@ -1,13 +1,121 @@
...
+        'markdown:Generate the markdown documentation.'
+        'md:Generate the markdown documentation.'
+        'config:Outputs a commented version of the configuration file that could be used
 by CRI-O. This allows you to save you current configuration setup and then load
-it later with **--config**. Global options will modify the output.' 'version:display detailed version information' 'wipe:wipe CRI-O's container and image storage' 'help:Shows a list of commands or help for one command' 'h:Shows a list of commands or help for one command')
+it later with **--config**. Global options will modify the output.'
+        'version:display detailed version information'
+        "wipe:wipe CRI-O's container and image storage"
+        'help:Shows a list of commands or help for one command'
+        'h:Shows a list of commands or help for one command')
```
[the remark this answers is not in the document]
Comment (TomSweeneyRedHat) [reply completions/zsh/_crio:15]: The same below too @fidencio ?

**Comment by TomSweeneyRedHat** [comment]
LGTM once tests are happy and @fidencitio suggestions are resolved.

File: completions/zsh/_crio [hunk 2]
```diff
@@ -1,13 +1,121 @@
 _cli_zsh_autocomplete() {
 
   local -a cmds
-  cmds=('complete:Generate bash, fish or zsh completions.' 'completion:Generate bash, fish or zsh completions.' 'man:Generate the man page documentation.' 'markdown:Generate the markdown documentation.' 'md:Generate the markdown documentation.' 'config:Outputs a commented version of the configuration file that could be used
+  cmds=('complete:Generate bash, fish or zsh completions.'
```
[the remark this answers is not in the document]
Comment (klihub) [reply completions/zsh/_crio]: Makes sense. I updated the implementation accordingly.

File: completions/zsh/_crio (hunk 1 above)
[the remark this answers is not in the document]
Comment (klihub) [reply completions/zsh/_crio]: Updated the implementation accordingly.

**Comment by fidencio** [comment]
LGTM, thanks @klihub!

**Comment by haircommander** [comment]
/retest-required

(if it continues to fail we can override e2e-agnositc, gcp passed)

ping @saschagrunert for final review/tag?

**Review by saschagrunert — APPROVED** [review]
/retest-required
/lgtm

**Pull request merged**